proxy/src/main/java/org/apache/rocketmq/proxy/service/route/MessageQueueSelector.java

```diff
@@ -154,6 +157,82 @@ public AddressableMessageQueue selectOne(boolean onlyBroker) {
         return selectOneByIndex(nextIndex, onlyBroker);
     }
 
+    public AddressableMessageQueue selectOneByPipeline(TopicRouteService topicRouteService, boolean onlyBroker) {
```

Comment:
It would be better to pass the TopicRouteService when constructing the MessageQueueSelector

Reply:
You are right. I have modified codes as you said. You can check the commit history.
Thank you~